tests/test_covariance_calculator.py

```diff
@@ -83,7 +83,11 @@ def test_create_sacc_cov():
     cc = CovarianceCalculator(input_yml)
```

Comment:
You could use a fixture here and use dependency injection to pass the cov calc into each test method.  

```
@pytest.fixture
def mock_covariance():
  return CovarianceCalculator(input_yml)

def my_test(mock_covariance):
   mock_covariance.get_covariance()
```
etc.  Definitely not required though, but something I came across when reading up on pytest.

Reply:
Interesting. I'll open an issue to improve the unit test with that but for now I'd rather touch as little as possible the actual code.